src/modules/commander/Commander.cpp

```diff
@@ -3022,125 +3182,7 @@ Commander::set_main_state_rc()
 			res = TRANSITION_NOT_CHANGED;
 
 		} else {
-			res = main_state_transition(_status, new_mode, _status_flags, &_internal_state);
-
-			/* ensure that the mode selection does not get stuck here */
-			int maxcount = 5;
-
-			/* enable the use of break */
-			/* fallback strategies, give the user the closest mode to what he wanted */
-			while (res == TRANSITION_DENIED && maxcount > 0) {
```

Comment:
I removed the while loop because i think that 
if a transition fails and we anyway don't wait for a change in the ekf state, just blindly re-trying will not improve things.
the re-tries were introduced to guarantee that the cycle would exit - getting rid of it completely still solves the problem